### PR TITLE
Add ability to assign value to OP_RETURN tx whilst building

### PR DESCRIPTION
**Pull request by zeptin**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Wallet/Models/RequestModels.cs [hunk 1]
```diff
@@ -225,6 +225,9 @@ public class BuildTransactionRequest : TxFeeEstimateRequest, IValidatableObject
 
         public string OpReturnData { get; set; }
 
+        [MoneyFormat(isRequired: false, ErrorMessage = "The op return value is not in the correct format.")]
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Wallet/Models/RequestModels.cs]:
```suggestion
         [MoneyFormat(isRequired: false, ErrorMessage = "The op return amount is not in the correct format.")]
```

File: src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs [hunk 2]
```diff
@@ -529,10 +529,15 @@ public TransactionBuildContext(Network network)
         public bool Shuffle { get; set; }
 
         /// <summary>
-        /// Optional data to be added as an extra OP_RETURN transaction output with Money.Zero value.
+        /// Optional data to be added as an extra OP_RETURN transaction output.
         /// </summary>
         public string OpReturnData { get; set; }
 
+        /// <summary>
+        /// Optional amount to add to the OP_RETURN transaction output to pass the stratisX standardness (dust) check.
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs]:
```suggestion
         /// Optional amount to add to the OP_RETURN transaction output.
```

What if someone just wants to burn some money?

**Comment by dangershony** [comment]
Did you make sure to enable none standard trx?

**Comment by zeptin** [comment]
> Did you make sure to enable none standard trx?

I don't think that's necessary, the goal of the integration test is to verify that stratisX accepts the transactions without additional tweaking.

**Pull request merged**